File: longsq/longsq_21/solution.go
package problem21

import (
	"encoding/json"
	. "leetCode/golang/models"
	"log"
	"strings"
)

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {

}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var list1 *ListNode
	var list2 *ListNode

	var list1IntArray []int
	if err := json.Unmarshal([]byte(inputValues[0]), &list1IntArray); err != nil {
		log.Fatal(err)
	}
	list1 = IntArrayToLinkedList(list1IntArray)
	var list2IntArray []int
	if err := json.Unmarshal([]byte(inputValues[1]), &list2IntArray); err != nil {
		log.Fatal(err)
	}
	list2 = IntArrayToLinkedList(list2IntArray)

	return mergeTwoLists(list1, list2).LinkedListToIntArray()
}

File: longsq/longsq_21/Solution.java
package longsq.longsq_21;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
import qubhjava.models.ListNode;

public class Solution extends BaseSolution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 == null || list2 == null) return list1 == null ? list2 : list1;
        ListNode head = new ListNode(-101);
        ListNode cur = head;
        while (list1 != null && list2 != null) {
            if (list1.val < list2.val) {
                cur.next = list1;
                cur = cur.next;
                list1 = list1.next;
                cur.next = null;
            } else {
                cur.next = list2;
                cur = cur.next;
                list2 = list2.next;
                cur.next = null;
            }
        }
        if (list1 != null) {
            cur.next = list1;
        } else {
            cur.next = list2;
        }
        return head.next;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        ListNode list1 = jsonArrayToListNode(inputJsonValues[0]);
		ListNode list2 = jsonArrayToListNode(inputJsonValues[1]);
        return JSON.toJSON(ListNode.LinkedListToIntArray(mergeTwoLists(list1, list2)));
    }
}
